src/Explorer/Notebook/NotebookContentClient.ts

```diff
@@ -194,18 +194,24 @@ export class NotebookContentClient {
       });
   }
 
-  public readFileContent(filePath: string): Promise<string> {
-    const fileType = NotebookUtil.isNotebookFile(filePath) ? "notebook" : "file";
-    return this.contentProvider
-      .get(this.getServerConfig(), filePath, { type: fileType, format: "text", content: 1 })
-      .toPromise()
-      .then(xhr => {
-        const content = (xhr.response as any).content;
-        if (!content) {
-          throw new Error("No content read");
-        }
+  public async readFileContent(filePath: string): Promise<string> {
```

Comment:
Not blocking, but why can't you using `fetch` and `response.blob()` for all of this? Can't Juno return a binary response?

Reply:
It's Nteract which is talking to Jupyter server and unfortunately we can't tell it to get a binary response currently. It'll be a bigger change to allow Nteract do that which we can tackle later on. This unblocks users right now who want to download binary files.